packages/SwingSet/misc-tools/replay-transcript.js

```diff
@@ -324,146 +324,24 @@ async function replay(transcriptFile) {
       bundleHandler,
     });
     factory = makeXsSubprocessFactory({
-      allVatPowers,
+      startXSnap,
       kernelKeeper: fakeKernelKeeper,
       kernelSlog,
-      startXSnap,
       testLog,
     });
   } else if (worker === 'local') {
     factory = makeLocalVatManagerFactory({
       allVatPowers,
-      kernelKeeper: fakeKernelKeeper,
```

Comment:
Why is it not needed for local but needed for xsnap worker?

Reply:
the xsnap worker is still doing:
```js
    const vatKeeper = kernelKeeper.provideVatKeeper(vatID);
    const snapshotInfo = vatKeeper.getSnapshotInfo();
```
and then uses `snapshotInfo` for two purposes:
* 1: give `startXSnap` a boolean `reload` flag, which makes it call `snapStore` for the snapshot contents
* 2: send a `setBundle` command to the worker only if there was no snapshot

Both should go away in a future PR where I change the manager-factory to have separate entrypoints for `initialize` and `loadSnapshot`. The latter will accept the snapshot reader stream, will pass it directly through to `startXSnap`, and won't need a `vatKeeper` to ask about snapshots at all.